genesis/sensors/base_sensor.py

```diff
@@ -162,31 +174,104 @@ def read_ground_truth(self, envs_idx: List[int] | None = None):
         """
         return self._get_formatted_data(self._manager.get_cloned_from_cache(self, is_ground_truth=True), envs_idx)
 
+    def add_recorder(self, handler: "DataHandler", rec_options: RecordingOptions | None = None):
+        """
+        Add a sensor data recorder that processes data from this sensor.
+        """
+        if self._data_recorder is None:
+            self._data_recorder = self._manager.create_data_recorder(self)
+        self._data_recorder.add_handler(handler, rec_options)
+
+    def start_recording(self):
+        """
+        Start recording the sensor data.
+        """
+        if self._data_recorder is None:
+            gs.raise_exception("Sensor data recorder not found. Call `add_recorder()` first.")
+        self._data_recorder.start()
+
+    def pause_recording(self):
+        """
+        Pause recording the sensor data.
+        """
+        if self._data_recorder is None:
+            gs.raise_exception("Sensor data recorder not found. Call `add_recorder()` first.")
+        self._data_recorder.pause()
+
+    def stop_recording(self):
+        """
+        Stop recording the sensor data.
+        """
+        if self._data_recorder is None:
+            gs.raise_exception("Sensor data recorder not found. Call `add_recorder()` first.")
+        self._data_recorder.stop()
+
+    @property
+    def is_built(self) -> bool:
+        return self._manager._sim._scene._is_built
+
+    # =============================== private shared methods ===============================
+
     @classmethod
     def _apply_delay_to_shared_cache(
-        self, shared_metadata: SharedSensorMetadata, shared_cache: torch.Tensor, buffered_data: "TensorRingBuffer"
+        cls,
+        shared_metadata: SharedSensorMetadata,
+        shared_cache: torch.Tensor,
+        buffered_data: "TensorRingBuffer",
+        jitter: torch.Tensor | None = None,
+        interpolate_for_delay: list[bool] | None = None,
     ):
         """
         Applies the read delay to the shared cache tensor by copying the buffered data at the appropriate index.
+
+        NOTE: This is a helper method for `update_shared_cache()` to apply the delay to the shared cache tensor.
```

Comment:
```
Note
----
This is a helper method for `update_shared_cache()` to apply the delay to the shared cache tensor.
```

By the way, what does it mean? You mean, "this method is used internally by `update_shared_cache`"? This should be sufficient. We already know that it applies the delay to the shared cache tensor from the short doc.